subprojects/launcher/src/main/java/org/gradle/launcher/daemon/client/DaemonClient.java

```diff
@@ -239,9 +243,23 @@ private Result handleDaemonDisappearance(Build build, DaemonDiagnostics diagnost
             + "\nAttempting to read last messages from the daemon log...", build);
 
         LOGGER.error(diagnostics.describe());
+        findCrashLogFile(build, diagnostics).ifPresent(crashLogFile ->
+            LOGGER.error("JVM crash log found: " + new ConsoleRenderer().asClickableFileUrl(crashLogFile))
+        );
         throw new DaemonDisappearedException();
     }
 
+    private Optional<File> findCrashLogFile(Build build, DaemonDiagnostics diagnostics) {
+        String crashLogFileName = "hs_err_pid" + diagnostics.getPid() + ".log";
+        List<File> candidates = ImmutableList.of(
+            new File(build.getParameters().getCurrentDir(), crashLogFileName),
+            new File(diagnostics.getDaemonLog().getParent(), crashLogFileName)
```

Comment:
We should also look in the temp directory according to https://stackoverflow.com/a/5154619/104894:

> Where possible the file is created in the working directory of the process. In the event that the file cannot be created in the working directory (insufficient space, permission problem, or other issue) **then the file is created in the temporary directory for the operating system.** On Solaris and Linux the temporary directory is /tmp. On Windows the temporary directory is specified by the value of the TMP environment variable, or if that is not defined, the value of the TEMP environment variable.

Reply:
I am looking at the temp directory as well. Please have another look!